Improve support for macOS

DamienCassou opened the pull request:
- Fix checking existence of the macOS files within the specified directory
- Make path to macOS scripts configurable
- Make path to macOS scripts if it doesn't exist yet

DamienCassou committed:
Check existence of the macOS files within the specified directory

By moving the configuration of `default-directory`, the
calls to `file-exists-p` will be executed within the expected
directory instead of in a random location.

DamienCassou committed:
Make path to macOS scripts configurable

This is useful if the directory containing emacs-everywhere's code
isn't writable: this happens when the Emacs package is bundled within
an OS package.

DamienCassou committed:
Make emacs-everywhere-osx-dir if it doesn't exist yet

tecosaur commented:
Ah yep, now I take more than a cursory glance at the changes, it looks fine to me :+1:

DamienCassou committed:
Improve default-value of emacs-everywhere--dir

When the file is loaded manually (e.g., through `eval-buffer`),
`load-file-name` is nil. This change makes it possible to get
a meaningful value even in this case.

DamienCassou commented on emacs-everywhere.el:
```diff
@@ -601,12 +601,15 @@ Please go to 'System Preferences > Security & Privacy > Privacy > Accessibility'
        :title window-title
        :geometry window-geometry))))
 
-(defvar emacs-everywhere--dir (file-name-directory load-file-name))
+(defcustom emacs-everywhere-osx-dir
+  (file-name-directory (or load-file-name buffer-file-name))
+  "Where macOS scripts are stored."
+  :type 'directory)
```
For (1), I moved the change to a dedicated commit explaining the rationale. This change makes it possible to use `M-x eval-buffer` to load `emacs-everywhere.el`.

For (2), the commit message already explains the rationale but I just added some information in the documentation of the variable. My package manager (nix) puts all Emacs packages in a non-writable directory so I have to change the value of this variable.

tecosaur replied:
That's completely fair. Thanks for your effort so far.

The one bit of extra effort I would like you to ask you for is what a good alternative directory to use would be? I'm happy to do a bit of the legwork with making this work, but I'm not familiar enough with MacOS to know what would be a good behaviour.

DamienCassou replied:
I would follow [no-littering](https://github.com/emacscollective/no-littering) convention and use something like `~/.emacs.d/var/emacs-everywhere/`.

DamienCassou commented:
I answered your feedback. Is there anything else I can do?

DamienCassou commented:
I would like to close this PR to cleanup my list of pending tasks.

DamienCassou commented:
I'm closing. Feel free to reopen if planning to work on it.

DamienCassou commented:
[Emacs-anywhere](https://github.com/nohzafk/emacs-anywhere) seems to work better than emacs-everywhere on macos.
